### PR TITLE
Prevent error when component is not found By.directive

### PR DESCRIPTION
**Pull request by MartinNuc**
Solves https://github.com/NetanelBasal/spectator/issues/85

### CODE REVIEW

File: projects/spectator/src/lib/errors.ts [hunk 1]
```diff
@@ -1 +1,14 @@
-export class SpectatorError extends Error {}
+export class SpectatorError extends Error {
+  constructor(message: string) {
+    super(message);
+    // https://github.com/Microsoft/TypeScript/wiki/Breaking-Changes#extending-built-ins-like-error-array-and-map-may-no-longer-work
+    Object.setPrototypeOf(this, SpectatorError.prototype);
+  }
+}
+export class SpectatorDebugElementNotFoundError extends SpectatorError {
```
Comment (MartinNuc) [inline projects/spectator/src/lib/errors.ts]: I created a specific error class for this case.

File: src/app/zippy/zippy.component.ts [hunk 2]
```diff
@@ -38,6 +33,11 @@ export class ZippyComponent {
   visible = false;
   updatedAsync = false;
 
+  @HostListener('keyup.esc')
+  onEsc() {
+    this.toggle();
+  }
```
Comment (MartinNuc) [inline src/app/zippy/zippy.component.ts]: linter complained here that method definitions are supposed to be after properties.

File: projects/spectator/src/lib/internals.ts [hunk 3]
```diff
@@ -70,7 +70,15 @@ export class Spectator<C> {
...
-    return _getChild<R>(this.debugElement)(directiveOrSelector, options);
+    try {
+      return _getChild<R>(this.debugElement)(directiveOrSelector, options);
+    } catch (err) {
+      if (err instanceof SpectatorDebugElementNotFoundError) {
+        return null;
+      } else {
+        throw err;
+      }
+    }
```
Comment (NetanelBasal) [inline projects/spectator/src/lib/internals.ts]: What about queryAll?

Comment (MartinNuc) [reply projects/spectator/src/lib/internals.ts]: Tested that one and it worked fine so no change is required.

Test for it is here: https://github.com/NetanelBasal/spectator/pull/86/files#diff-e6572426b83a2437724cd9a71d19a7deR92

**Pull request merged**